python/pyspark/pandas/indexes/base.py

```diff
@@ -1917,18 +1917,12 @@ def append(self, other: "Index") -> "Index":
         sdf_other = other._internal.spark_frame.select(other._internal.index_spark_columns)
         sdf_appended = sdf_self.union(sdf_other)
 
-        # names should be kept when MultiIndex, but Index wouldn't keep its name.
-        if isinstance(self, MultiIndex):
-            index_names = self._internal.index_names
-        else:
-            index_names = None
-
         internal = InternalFrame(
             spark_frame=sdf_appended,
             index_spark_columns=[
                 scol_for(sdf_appended, col) for col in self._internal.index_spark_column_names
             ],
-            index_names=index_names,
+            index_names=None,
```

Comment:
We can simply set the `index_names` to `None` to follow the behavior of Pandas, since Pandas doesn't keep the name of `MultiIndex` when computing the `append` from Pandas 2.0.0. (See https://github.com/pandas-dev/pandas/pull/48288 more detail)

cc @zhengruifeng @HyukjinKwon as CI passed.

Reply:
is it a bug in Pandas that might be fixed in the future?

Reply:
I believe it's an intentional behavior since they mentioned this in ["Bug fixes"](https://pandas.pydata.org/docs/dev/whatsnew/v2.0.0.html#bug-fixes) section in their release note?

Reply:
so I thought it was a bug in Pandas that is fixed in the Pandas 2.0.0.

Reply:
shouldn't we also mention this in our migration doc?

Reply:
ok, I misunderstood the Pandas PR. LTGM

Reply:
> shouldn't we also mention this in our migration doc?

Hmm.. I didn't mention this as a behavior change since it's a bug fix, but on second thought maybe we'd better to mention in the migration guide anyway.

Let me create a follow-up for updating the migration guide.